src/applications/model/gaming-streaming-server.h

```diff
@@ -0,0 +1,149 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors: Salman Mohebi <s.mohebi22@gmail.com>
+ *
+ */
+
+#ifndef GAMING_STREAMING_SERVER_H
+#define GAMING_STREAMING_SERVER_H
+
+#include "ns3/application.h"
+#include "ns3/random-variable-stream.h"
+
+namespace ns3 {
+
+class Socket;
+/**
+ * \ingroup applications
+ *
+ * This traffic generator Generates different traffic streams
+ * based on different random variables
+ *
+ * Each random stream defined by two random variables:
+ *      packetSize: A random variable which determines the size of generated packets
+ *      interArrivalTime: A random variable which determines the packets inter-arrival time
+*/
+class GamingStreamingServer : public Application
+{
+public:
+  /**
+   * \brief Get the type ID.
+   * \return the object TypeId
+   */
+  static TypeId GetTypeId (void);
+
+  /**
+   * \brief create a GamingStreamingServer object with default parameters
+   */
+  GamingStreamingServer ();
+
+  /**
+   * \brief Create a GamingStreamingServer object
+   *
+   * \param ip Remote peer address
+   * \param port Remote peer port
+   */
+  GamingStreamingServer (Ipv4Address ip, uint16_t port);
+  virtual ~GamingStreamingServer ();
+
+  /**
+   * \brief set the remote address and port
+   *
+   * \param ip remote IP address
+   * \param port remote port
+   */
+  void SetRemote (Address ip, uint16_t port);
+
+  /**
+   * \brief set the remote address
+   * \param addr remote address
+   */
+  void SetRemote (Address addr);
+
+  /**
+   * \brief return the number of total sent packets
+   * \return total number of sent packets
+   */
+  uint32_t GetTotSentPackets (void);
+
+  /**
+   * \brief return the total bytes sent
+   * \return the total bytes sent
+   */
+  uint32_t GetTotSentBytes (void);
+
+  /**
+   * \brief Erase the statistics of sent packets
+   */
+  void EraseStatistics (void);
+
+  /**
+   * \brief Add parameters of new stream
+   *
+   * \param packetSize A random number generator for packet size
+   * \param interArrivalTime A random number generator for inter-arrival time
+   */
+  void AddNewTraffic (Ptr<RandomVariableStream> packetSize, Ptr<RandomVariableStream> interArrivalTime);
+
+  virtual void StartApplication (void);
+  virtual void StopApplication (void);
+
+protected:
+  /**
+   * \brief Generate and send packets based on the different random
+   * distribution for packet sizes and inter-arrival times
+   *
+   * \param idx index of stream within the streams vector
+   */
+  void Send (int idx);
+
+  /**
+   *  Initialize the parameters of different streams
+   */
+  virtual void InitializeStreams (void);
```

Comment:
`InitializeStreams` should be pure virtual. Accordingly, remove it from` GamingStreamingServer.cc`